hadoop-yarn-project/hadoop-yarn/hadoop-yarn-server/hadoop-yarn-server-resourcemanager/src/main/java/org/apache/hadoop/yarn/server/resourcemanager/scheduler/QueueMetrics.java

```diff
@@ -117,6 +117,8 @@ public class QueueMetrics implements MetricsSource {
   @Metric("Reserved CPU in virtual cores") MutableGaugeInt reservedVCores;
   @Metric("# of reserved containers") MutableGaugeInt reservedContainers;
 
+  public static final String CONFIGURATION_VALIDATION = "yarn.configuration-validation";
```

Comment:
I would prefer having this in the CapacitySchedulerConfiguration object (possibly with getters and a short description what is it used for), preferably with the prefix of CS, as this is irrelevant in FS cases.

Actually: do we need to have this as a config property? AFAIU we shouldn't set this via the xml, couldn't CS simply have a validation flag?

Reply:
At first I put it in the `CapacitySchedulerConfiguration`, but then I realised that the `QueueMetrics` (not the `CSQueueMetrics`) should not depend on anything from CS.

We don't really need a config property, and it should not be a public facing property either. But the Configuration object is shared and propagated everywhere were we need to check if we are in validation mode (`QueueMetrics` for the `PartitionQueueMetrics` and `CSQueueMetrics`). Not sure where else to propagate this flag, without introducing a dependency in `QueueMetrics` to `CapacitySechduler` related logic.